### ISSUE TITLE
[Remote-SSH Bug]: stuck in "Setting up SSH tunnel" after upgrading to 9/7 release

### ISSUE THREAD

**Issue by shiltian** [microsoft/vscode-remote-release#8925]
### Is there an existing issue for this bug?

- [X] I have searched the existing issues

### Required Troubleshooting Steps

- [X] I have followed these troubleshooting steps
- [X] I have tried both values of the `remote.SSH.useLocalServer` setting

### Connect Locally

It connects successfully

### Expected Behavior

It is expected to open the remote directory.

### Remote-SSH Log

<details>
<summary>Remote-SSH Log</summary>
<p>

```
[16:21:16.384] Log Level: 2
[16:21:16.390] SSH Resolver called for "ssh-remote+jlse", attempt 1
[16:21:16.390] "remote.SSH.useLocalServer": true
[16:21:16.390] "remote.SSH.path": undefined
[16:21:16.390] "remote.SSH.configFile": undefined
[16:21:16.390] "remote.SSH.useFlock": true
[16:21:16.390] "remote.SSH.lockfilesInTmp": false
[16:21:16.390] "remote.SSH.localServerDownload": auto
[16:21:16.390] "remote.SSH.remoteServerListenOnSocket": false
[16:21:16.390] "remote.SSH.showLoginTerminal": false
[16:21:16.390] "remote.SSH.defaultExtensions": []
[16:21:16.390] "remote.SSH.loglevel": 2
[16:21:16.390] "remote.SSH.enableDynamicForwarding": true
[16:21:16.390] "remote.SSH.enableRemoteCommand": false
[16:21:16.390] "remote.SSH.serverPickPortsFromRange": {}
[16:21:16.390] "remote.SSH.serverInstallPath": {}
[16:21:16.392] VS Code version: 1.82.0
[16:21:16.392] Remote-SSH version: remote-ssh@0.106.0
[16:21:16.392] darwin arm64
[16:21:16.392] SSH Resolver called for host: jlse
[16:21:16.392] Setting up SSH remote "jlse"
[16:21:16.394] Acquiring local install lock: /var/folders/gt/rdf1gsc11rn0k65kr2l6z94w0000gn/T/vscode-remote-ssh-71466f2f-install.lock
[16:21:16.394] Looking for existing server data file at /Users/shiltian/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-71466f2f-8b617bd08fd9e3fc94d14adb8d358b56e3f72314-0.106.0/data.json
[16:21:16.394] Using commit id "8b617bd08fd9e3fc94d14adb8d358b56e3f72314" and quality "stable" for server
[16:21:16.396] Install and start server if needed
[16:21:16.398] PATH: /Users/shiltian/.local/modules-5.3.1/bin:/Users/shiltian/.local/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin://Applications/Topaz Photo AI.app/Contents/Resources/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin
[16:21:16.398] Checking ssh with "ssh -V"
[16:21:16.402] > OpenSSH_9.0p1, LibreSSL 3.3.6

[16:21:16.404] askpass server listening on /var/folders/gt/rdf1gsc11rn0k65kr2l6z94w0000gn/T/vscode-ssh-askpass-5cfa0735770e4b08fd47b4d05904e1a696590ca1.sock
[16:21:16.404] Spawning local server with {"serverId":1,"ipcHandlePath":"/var/folders/gt/rdf1gsc11rn0k65kr2l6z94w0000gn/T/vscode-ssh-askpass-519d45d2b0f8fd668c89d902eaff5269dea5d0c3.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","49744","-o","ConnectTimeout=30","jlse","bash"],"serverDataFolderName":".vscode-server","dataFilePath":"/Users/shiltian/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-71466f2f-8b617bd08fd9e3fc94d14adb8d358b56e3f72314-0.106.0/data.json"}
[16:21:16.404] Local server env: {"SSH_AUTH_SOCK":"/private/tmp/com.apple.launchd.nLP5saVOhd/Listeners","SHELL":"/bin/zsh","DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/Users/shiltian/.vscode/extensions/ms-vscode-remote.remote-ssh-0.106.0/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin)","VSCODE_SSH_ASKPASS_EXTRA_ARGS":"--ms-enable-electron-run-as-node","VSCODE_SSH_ASKPASS_MAIN":"/Users/shiltian/.vscode/extensions/ms-vscode-remote.remote-ssh-0.106.0/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/var/folders/gt/rdf1gsc11rn0k65kr2l6z94w0000gn/T/vscode-ssh-askpass-5cfa0735770e4b08fd47b4d05904e1a696590ca1.sock"}
[16:21:16.404] Spawned 1756
[16:21:16.470] > local-server-1> Running ssh connection command: "-v -T -D 49744 -o ConnectTimeout=30 jlse bash"
[16:21:16.471] > local-server-1> Spawned ssh, pid=1763
[16:21:16.474] stderr> OpenSSH_9.0p1, LibreSSL 3.3.6
[16:21:16.707] stderr> debug1: Server host key: ssh-ed25519 SHA256:BNt1t4QjgYo9aS4df1wbEAZE5Dv9N6sJmT9D9foPQy0
[16:21:16.790] stderr> ---------------------------------------------------------------------------
[16:21:16.790] stderr>                             Notice to Users
[16:21:16.790] stderr> 
[16:21:16.790] stderr> This is a Federal computer system and is the property of the United States
[16:21:16.790] stderr> Government. It is for authorized use only. Users (authorized or unauthorized)
[16:21:16.790] stderr> have no explicit or implicit expectation of privacy.
[16:21:16.790] stderr> 
[16:21:16.790] stderr> Any or all uses of this system and all files on this system may be intercepted,
[16:21:16.790] stderr> monitored, recorded, copied, audited, inspected, and disclosed to authorized
[16:21:16.790] stderr> site, Department of Energy, and law enforcement personnel, as well as
[16:21:16.790] stderr> authorized officials of other agencies, both domestic and foreign. By using
[16:21:16.790] stderr> this system, the user consents to such interception, monitoring, recording,
[16:21:16.790] stderr> copying, auditing, inspection, and disclosure at the discretion of authorized
[16:21:16.790] stderr> site or Department of Energy personnel.
[16:21:16.790] stderr> 
[16:21:16.790] stderr> Unauthorized or improper use of this system may result in administrative
[16:21:16.790] stderr> disciplinary action and civil and criminal penalties. By continuing to use
[16:21:16.791] stderr> this system you indicate your awareness of and consent to these terms and
[16:21:16.791] stderr> conditions of use. LOG OFF IMMEDIATELY if you do not agree to the conditions
[16:21:16.791] stderr> stated in this warning.
[16:21:16.791] stderr> 
[16:21:16.791] stderr> ----------------------------------------------------------------------------
[16:21:16.791] stderr> 
[16:21:16.791] stderr> Duo Authentication Required.
[16:21:16.791] stderr> 
[16:21:16.791] stderr> ----------------------------------------------------------------------------
[16:21:16.891] stderr> Authenticated using "publickey" with partial success.
[16:21:17.492] Got askpass request: {"request":"(username@hostname) Duo two-factor login for username Enter a passcode or select one of the following options: 1. Duo Push to XXX-XXX-0814 2. Duo Push to iPad Pro (iOS) Passcode or option (1-2):"}
[16:21:17.493] Listening for interwindow password on /var/folders/gt/rdf1gsc11rn0k65kr2l6z94w0000gn/T/vscode-ssh-askpass-661e07ff73c6126544e2d704ebe575f84f90618a.sock
[16:21:17.493] Writing password prompt to globalState
[16:21:19.193] Interactor gave response: *
[16:21:19.194] Cleaning up other-window auth server
[16:21:23.545] stderr> Success. Logging you in...
[16:21:23.567] stderr> Authenticated to hostname ([remote-ip]:22) using "keyboard-interactive".
[16:21:24.142] > ready: 4647db9ed242
[16:21:24.167] > Linux 5.14.21-150400.24.46-default #1 SMP PREEMPT_DYNAMIC Thu Feb 9 08:38:18 UTC 2023 (2d95137)
[16:21:24.167] Platform: linux
[16:21:24.190] > /bin/zsh
[16:21:24.190] Shell: zsh
[16:21:24.190] Creating bash subshell inside "zsh"
[16:21:24.214] > bash version:  4.4.23(1)-release
[16:21:24.242] > 4647db9ed242: running
[16:21:24.273] > Found existing installation at /home/username/.vscode-server...
> Running ssh connection command... /home/username/.vscode-server/code-8b617bd08fd9e3fc94d14adb8d358b56e3f72314 command-shell --cli-data-dir /home/username/.vscode-server/cli --on-port --require-token 6e46bfac0588 &> "/home/username/.vscode-server/.cli.8b617bd08fd9e3fc94d14adb8d358b56e3f72314.log" < /dev/null
> printenv:
[16:21:24.275] >     SSH_CONNECTION=local-ip 49745 remote-ip 22
>     OLDPWD=/home/username
>     XDG_SESSION_ID=689
>     USER=username
>     PWD=/home/username
>     HOME=/home/username
>     SSH_CLIENT=local-ip 49745 22
>     XDG_SESSION_TYPE=tty
>     SHELL=/bin/zsh
>     XDG_SESSION_CLASS=user
>     SHLVL=2
>     VSCODE_AGENT_FOLDER=/home/username/.vscode-server
>     LOGNAME=username
>     DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/21198/bus
>     XDG_RUNTIME_DIR=/run/user/21198
>     PATH=/usr/bin:/bin:/usr/sbin:/sbin
>     _=/usr/bin/printenv
[16:21:24.277] > Removing old logfile at /home/username/.vscode-server/.cli.8b617bd08fd9e3fc94d14adb8d358b56e3f72314.log
[16:21:24.280] > Spawned remote CLI: 72462
[16:21:24.283] > Waiting for server log...
[16:21:24.317] > Waiting for server log...
[16:21:24.351] > Waiting for server log...
[16:21:24.391] > 4647db9ed242: start
> SSH_AUTH_SOCK====
> DISPLAY====
> listeningOn==127.0.0.1:45461==
> osReleaseId==opensuse-leap==
> arch==x86_64==
> vscodeArch==x64==
> bitness==64==
> tmpDir==/run/user/21198==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> serverStartTime==111==
> 4647db9ed242: end
[16:21:24.391] Received install output: 
SSH_AUTH_SOCK====
DISPLAY====
listeningOn==127.0.0.1:45461==
osReleaseId==opensuse-leap==
arch==x86_64==
vscodeArch==x64==
bitness==64==
tmpDir==/run/user/21198==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
serverStartTime==111==

[16:21:24.391] Remote server is listening on socket 127.0.0.1:45461
[16:21:24.391] Parsed server configuration: {"serverConfiguration":{"remoteListeningOn":{"socketPath":"127.0.0.1:45461"},"osReleaseId":"opensuse-leap","arch":"x86_64","sshAuthSock":"","display":"","tmpDir":"/run/user/21198","platform":"linux"},"serverStartTime":111,"installUnpackCode":""}
[16:21:24.392] Persisting server connection details to /Users/shiltian/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-71466f2f-8b617bd08fd9e3fc94d14adb8d358b56e3f72314-0.106.0/data.json
[16:21:24.394] Running script with connection command: ssh -T -L 127.0.0.1:49747:127.0.0.1:45461 -o ConnectTimeout=30 jlse bash
[16:21:24.394] Spawning tunnel with: ssh -T -L 127.0.0.1:49747:127.0.0.1:45461 -o ConnectTimeout=30 jlse bash << 'EOSSH'
echo $SSH_AUTH_SOCK && echo -e 'Connected to SSH Host - Please do not close this terminal' && while true; do sleep 180; echo -n ' '; done
EOSSH
[16:21:24.884] > ---------------------------------------------------------------------------
>                             Notice to Users
> 
> This is a Federal computer system and is the property of the United States
> Government. It is for authorized use only. Users (authorized or unauthorized)
> have no explicit or implicit expectation of privacy.
> 
> Any or all uses of this system and all files on this system may be intercepted,
> monitored, recorded, copied, audited, inspected, and disclosed to authorized
> site, Department of Energy, and law enforcement personnel, as well as
> authorized officials of other agencies, both domestic and foreign. By using
> this system, the user consents to such interception, monitoring, recording,
> copying, auditing, inspection, and disclosure at the discretion of authorized
> site or Department of Energy personnel.
> 
> Unauthorized or improper use of this system may result in administrative
> disciplinary action and civil and criminal penalties. By continuing to use
> this system you indicate your awareness of and consent to these terms and
> conditions of use. LOG OFF IMMEDIATELY if you do not agree to the conditions
> stated in this warning.
> 
> ----------------------------------------------------------------------------
> 
> Duo Authentication Required.
> 
> ----------------------------------------------------------------------------
[16:21:25.413] > (username@hostname) Duo two-factor login for username
> 
> Enter a passcode or select one of the following options:
> 
>  1. Duo Push to XXX-XXX-0814
>  2. Duo Push to iPad Pro (iOS)
> 
> Passcode or option (1-2): 
```

</p>
</details>

### Anything else?

For the security reason, I have hidden the user name, host name, and ip address in the log above.

We have DUO in the remote system. Every time after I start the connect, it pops up a dialog to ask me to choose the authentication device, as shown in the log above (the "Enter a passcode or select one of the following options" part). I just need to input 1 and then press enter, and then everything is supposed to move forward. However, now it still pops up the dialog, and I still input 1, but after that, it is stuck at "Setting up SSH tunnel".

I can successfully ssh to the remote system using the "Running SSH connection command" above and execute commands.
```
...
debug1: client_global_hostkeys_private_confirm: server used untrusted RSA signature algorithm ssh-rsa for key 0, disregarding
debug1: update_known_hosts: known hosts file /Users/shiltian/.ssh/known_hosts2 does not exist
ls <--- This is the command I typed in
Documents
Downloads
lttng-traces
privatemodules
```

Please let me know if anything else is needed.

**Comment by shiltian** [comment]
What's more, I can start the vscode server manually and then use `https://vscode.dev/tunnel/...` in the browser to access the remote system.

**Comment by adrifoster** [comment]
I am also getting this issue

**Comment by yuichiok** [comment]
same here

**Comment by adrifoster** [comment]
I actually got this to work by setting up the suggested settings in the Troubleshooting section:

```
"remote.SSH.showLoginTerminal": true,
"remote.SSH.useLocalServer": false
```

But it is behaving very oddly. It seems to need my pw and Duo authentication **twice** for connecting, choosing a folder, etc. I'm not sure why this is happening?

**Comment by shiltian** [comment]
> I actually got this to work by setting up the suggested settings in the Troubleshooting section:
> 
> ```
> "remote.SSH.showLoginTerminal": true,
> "remote.SSH.useLocalServer": false
> ```
> 
> But it is behaving very oddly. It seems to need my pw and Duo authentication **twice** for connecting, choosing a folder, etc. I'm not sure why this is happening?

That doesn't work on my side. It pops out authentication twice but after confirming the second one the connection is closed immediately.

**Comment by roblourens** [comment]
Sorry, try setting `"remote.SSH.useExecServer": false` to work around it. Fix on the way

**Comment by yuichiok** [comment]
Yes, works for me.

**Comment by connor4312** [comment]
Thanks all, a fix has been published in remote-ssh 0.106.1. Coming soon to a CDN near you™️
